josue/js/app.js

```diff
@@ -8,74 +8,104 @@ const registerFieldsWrapper = document.querySelector("#register-fields");
     const newFilteredFormArray = [...new Set(fieldsCombined)]
     validator.fields = newFilteredFormArray;
-    registerFieldsIds.forEach(field=>{
+    registerFieldsIds.forEach(field => {
         const input = document.querySelector(`#${field}`);
-        input.addEventListener('input',()=>{
+        input.addEventListener('input', () => {
             validator.validateFields(input);
         });
     });
 }
-const parseFormData= ()=>{
+const parseFormData = () => {
     const myObjectFromForm = {}
     validator.fields.forEach(field => {
         const input = document.querySelector(`#${field}`)
         myObjectFromForm[field] = input.value;
     })
+    myObjectFromForm["balance"] = 0;
     return myObjectFromForm
 }
 
-signInBtn.addEventListener("click",()=>{
+signInBtn.addEventListener("click", () => {
     registerFieldsWrapper.remove();
     submitBtn.innerText = SIGN_IN;
-    validator.fields = validator.fields.slice(0,2);
+    validator.fields = validator.fields.slice(0, 2);
 });
 
-registerBtn.addEventListener("click",()=>{
-    form.insertBefore(registerFieldsWrapper,submitBtn);
+registerBtn.addEventListener("click", () => {
+    form.insertBefore(registerFieldsWrapper, submitBtn);
     submitBtn.innerText = REGISTER;
     addRegisterFieldsToValidator();
 });
 
-submitBtn.addEventListener('click',e=>{
+submitBtn.addEventListener('click', e => {
     e.preventDefault();
     const isFormValid = validator.validateOnSubmit();
-    if(isFormValid){
+    if (isFormValid) {
         // Fields are filled correctly now we check if user wants to register or login
-        if(submitBtn.innerText == SIGN_IN){
+        if (submitBtn.innerText == SIGN_IN) {
             const userData = parseFormData();
-            const user = JSON.parse(window.localStorage.getItem(`${userData.phone}`));
-            console.log(user);
-            if(user){
-                welcomeSplash.remove();
-                accountSection.appendChild(boardWrapper);
-                userDashboard.firstElementChild.innerHTML = `Welcome ${user.email}`;
-            }else{
-                alert("You are not part of BankApp, want to register instead?");
-            }
+            USER = getUserFromLocalStorage(userData);
+            USER ? loadUserDashboard() : alert("You are not an user, you have to register first!");
         }
-        else{
+        else {
             // User wants to register.. save it to localstorage
-            const  userData = parseFormData();
-            window.localStorage.setItem(`${userData.phone}`,JSON.stringify(userData));
+            const userData = parseFormData();
+            window.localStorage.setItem(`${userData.phone}`, JSON.stringify(userData));
             confirm("You are now part of BankApp go to Sign In!");
         }
     }
 });
+const getUserFromLocalStorage = userData => {
+    const user = JSON.parse(window.localStorage.getItem(`${userData.phone}`));
```

Comment:
NIT: `window.localStorage.getItem(`${userData.phone}`)` return value should be in its own variable